source/Application/Controller/ArticleDetailsController.php

```diff
@@ -1542,12 +1528,7 @@ public function getDefaultSorting()
      */
     public function getSortingParameters()
     {
-        $sorting = $this->getSorting($this->getSortIdent());
-        if (!is_array($sorting)) {
-            return null;
-        }
-
-        return implode('|', $sorting);
+        return implode('|', (array) $this->getSorting($this->getSortIdent())) ?: null;
```

Comment:
Personally for me there is a need to look at this line for 30 seconds till I get what it does. I would suggest to split logic here.